test/parallel/test-buffer.js

```diff
@@ -1476,3 +1476,26 @@ assert.equal(SlowBuffer.prototype.offset, undefined);
   // Check pool offset after that by trying to write string into the pool.
   assert.doesNotThrow(() => Buffer.from('abc'));
 }
+
+
+// Test failed or zero-sized Buffer allocations not affecting typed arrays
+{
+  const zeroArray = new Uint32Array(10).fill(0);
```

Comment:
is the `.fill(0)` just precautionary in case it hasn't been properly zero filled?